components/eam/src/physics/crm/crm_physics.F90

```diff
@@ -1001,6 +1090,15 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
          ptend%q(1:ncol,1:pver,idx_vt_q) = crm_output%q_vt_tend(1:ncol,1:pver)
       end if
 
+      !---------------------------------------------------------------------------------------------
+      ! set convective heating tendency for gravity wave drag
+      !---------------------------------------------------------------------------------------------
+      ttend_dp_idx = pbuf_get_index('TTEND_DP')
+      if (ttend_dp_idx > 0) then
+         call pbuf_get_field(pbuf, ttend_dp_idx, ttend_dp)
+         ttend_dp(:ncol,:pver) = ptend%s(:ncol,:pver)/cpair
```

Comment:
ttend_dp is completely overwritten here. Why do we need to read ttend_dp from pbuf? Doesn't it have to be pbuf_set_field after line 1099?

Reply:
We want to overwrite it here. This is the heating that is used to calculate the convective gravity wave source, and since the CRM is providing all the convective heating this is the only place that it needs to be set. 

AFAIK, using ttend_dp as a pointer in this way is identical to using pbuf_set_field()